.github/workflows/jest-tests.yml

```diff
@@ -0,0 +1,31 @@
+name: Jest Tests
+
+on:
+  pull_request:
+
+jobs:
+  jest-tests:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: '18'
+
+      - name: Cache Node.js modules
+        uses: actions/cache@v4
+        with:
+          path: ~/.npm
+          key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-node-
+
+      - name: Install dependencies
+        run: npm install && npm run build
+
+      - name: Run tests
+        run: npm test
```

Comment:
@ChrisWeissmann dont we need like `npm run test`? It may be good to have everything reference the single script/command in package.json that way if you ever need to make a change, you only need to adjust that one place... what do you think?